utils_BAP.py

```diff
@@ -0,0 +1,202 @@
+    """
+    Calculate date score from SCL data.
+
+    Args:
+        scl (DataCube): SCL data cube.
+
+    Returns:
+        DataCube: Date score data cube.
+    """
+    _log.info(f'calculating date score')
+    # Calculate date score
+    day_of_month = scl.apply_neighborhood(
+        day_of_month_calc,
+        size=[{'dimension': 'x', 'unit': 'px', 'value': 1},
+              {'dimension': 'y', 'unit': 'px', 'value': 1},
+              {'dimension': 't', 'value': "month"}],
+        overlap=[]
+    )
+    date_score = (1.0 * day_of_month).apply(date_score_calc)
+    date_score = date_score.rename_labels('bands', ['score'])
+    return date_score
+
+def calculate_distance_to_cloud_score(binary: openeo.DataCube,
+                                      spatial_resolution: int) -> openeo.DataCube:
+    """
+    Calculate distance to cloud score.
+
+    Args:
+        binary (DataCube): Binary cloud mask data cube.
+        spatial_resolution (int): Spatial resolution.
+
+    Returns:
+        DataCube: Distance to cloud score data cube.
+    """
+    _log.info(f'calculating distance to cloud score')
+    # Calculate dtc score
+    kernel_size = int(150 * 20 / spatial_resolution) + 1
+    gaussian_1d = gaussian(M=kernel_size, std=1)
+    gaussian_kernel = np.outer(gaussian_1d, gaussian_1d)
+    gaussian_kernel /= gaussian_kernel.sum()
+    dtc_score = 1 - binary.apply_kernel(gaussian_kernel)
+    return dtc_score
+
+def create_rank_mask(score: openeo.DataCube) -> openeo.DataCube:
+    """
+    Create a rank mask based on the input score.
+
+    Args:
+        score (DataCube): Input score data cube.
+
+    Returns:
+        DataCube: Rank mask data cube.
+    """
+    _log.info(f'calculating rank mask')
+    # Create a rank mask
+    rank_mask = score.apply_neighborhood(
+        max_score_selection,
+        size=[{'dimension': 'x', 'unit': 'px', 'value': 1},
+              {'dimension': 'y', 'unit': 'px', 'value': 1},
+              {'dimension': 't', 'value': "month"}],
+        overlap=[]
+    )
+    rank_mask = rank_mask.band('score')
+    return rank_mask
+
+def aggregate_BAP_scores(dtc_score: openeo.DataCube,
+                        date_score: openeo.DataCube, 
+                        coverage_score: openeo.DataCube) -> openeo.DataCube:
+    """
+    Aggregate BAP scores using weighted sum.
+
+    Args:
+        dtc_score (DataCube): Distance to cloud score data cube.
+        date_score (DataCube): Date score data cube.
+        coverage_score (DataCube): Cloud coverage score data cube.
+
+    Returns:
+        DataCube: Aggregated BAP score data cube.
+    """
+    _log.info(f'aggregating rank score')
+    # Aggregate scores
+    weights = [1, 0.8, 0.5]
+    score = (weights[0] * dtc_score +
+             weights[1] * date_score +
+             weights[2] * coverage_score) / sum(weights)
+    return score
+
+
+def create_rank_mask(score: openeo.DataCube) -> openeo.DataCube:
```

Comment:
You have create_rank_mask defined twice.

Reply:
will remove